### ISSUE TITLE
Disable prompt character

### ISSUE THREAD

**Issue by PoisonFrog** [Rolv-Apneseth/starship.yazi#8]
Is it possible to make starship.yazi disable the character module without using a whole new configuration file that has it disabled? I just don't want to see the prompt icon ("❯") at the top.

**Comment by Rolv-Apneseth** [comment]
Hmm - I'll have a look but I'm not sure. Of course if it was on 2 lines it would just be cut off but I understand some prefer to keep it on one line.

I know you can do `starship toggle character`, but the user needs to have the `disabled` line in their config, and it feels wrong to be messing with the user config.

Maybe `starship module character`, measure the length of that and cut it off the end of the string. I'll look into it when I have some time for this

**Comment by Rolv-Apneseth** [comment]
Yeah not sure how to do this, I think some kind of toggle needs to be added to `starship` first. I tried the second approach I mentioned but it seems there are some inconsistencies in the length of the prompt icon so as far as I can tell that doesn't seem to work reliably.

I see a couple of the config options in `starship` have `detect_env_vars` which could potentially be helpful if it existed for `character`.

I'll leave this open in case there are any updates to `starship`, or if anyone else wants to try tackle this, my attempt is on [this branch](https://github.com/Rolv-Apneseth/starship.yazi/tree/option_disable_prompt).

**Comment by PoisonFrog** [comment]
I see. Do you think it would be possible to make the config_file property accept a relative path? This would at least make the plugin configuration more portable when using a custom config.

**Comment by Rolv-Apneseth** [comment]
The issue is it's not running in a shell so it doesn't expand `~` and `$HOME`. I can however implement a manual conversion of those in the given path to the actual home directory by fetching the OS variable `$HOME` separate. Does seem a bit jank though, what do you think?

**Comment by PoisonFrog** [comment]
> Does seem a bit jank though, what do you think?

I'm not very technical. If it is a janky solution then maybe you should pass on it, after all this is just an inconvenience and not a breaking bug or anything like that.

**Comment by Rolv-Apneseth** [comment]
No worries. I'll just implement it cause I don't think it's too bad

**Comment by Rolv-Apneseth** [comment]
There you are, #9 is merged so you should be able to use `~` or `$HOME`, let me know if it works for you

**Comment by PoisonFrog** [comment]
> There you are, #9 is merged so you should be able to use `~` or `$HOME`, let me know if it works for you

Yes, it is relative paths are working now!
Thank you!

**Comment by blackdaemon** [comment]
I was digging into starship docs and found possible ways to hide the prompt character, might be helpful to you:

- That prompt symbol is called 'character' in the starship terminology

- We can toggle the `character~ before displaying the prompt by calling `starship toggle character`, however this has a potential to do the opposite for people having the 'character' off by default

- Make a temporary copy of starship.toml config file if it exists or create new one and modify it or add new setting to switch off the 'character': 
```
[character]
disabled=true
```
and use that temporary config (by overriding the env variable) when calling `starship prompt`

**Comment by Rolv-Apneseth** [comment]
Thanks for the suggestions. I believe that would be a bit too much effort for simply disabling the prompt character, though if you want to have a go at an implementation, I could consider it.

The solution I was hoping for is using the `detect_env_vars` that is available for many other `starship` config sections to disable the character ([like here for the hostname](https://starship.rs/config/#hide-the-hostname-in-remote-tmux-sessions)), though it seems that is not yet an option.

Or maybe if [this pr](https://github.com/starship/starship/pull/4439) got merged.
